### PR TITLE
Refactored injecting GUIDs as the code list ids for data criteria int…

### PR DESCRIPTION
**Pull request by jbradl11**
…o a separate function

Pull requests into Bonnie Bundler require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

Refactored code that adds code_list_ids to the Measure's source data criteria and data criteria into a separate function. This will allow us to easily call this function in our "rebuild_elm" script in Bonnie.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR:  https://jira.mitre.org/browse/BONNIE-1242
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced. N/A
- [x] Tests are included and test edge cases. 
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Review by lukoktonos — DISMISSED** [review]

File: lib/measures/loading/cql_loader.rb [hunk 1]
```diff
@@ -245,17 +251,18 @@ def self.generate_single_code_references(elms, all_codes_and_code_names, user)
 
             code_sets[code_system_name] ||= []
             code_sets[code_system_name] << code_reference['id']
-            # Generate a unique number as our fake "oid"
-            code_guid = SecureRandom.uuid
-            # Keep a list of generated_guids and a hash of guids with code system names and codes.
+            # Generate a unique number as our fake "oid" based on parameters that identify the DRC
+            code_guid = "drc-" + Digest::SHA2.hexdigest("#{code_system_name} #{code_reference['id']} #{code_reference['name']}")            # Keep a list of generated_guids and a hash of guids with code system names and codes.
```
Comment (hossenlopp) [inline lib/measures/loading/cql_loader.rb]: `code_guid` is now a misnomer. `code_hash` would be more appropriate.

File: lib/measures/loading/cql_loader.rb [hunk 2]
```diff
@@ -42,11 +42,28 @@ def self.load_mat_cql_exports(user, zip_file, out_dir, measure_details, vsac_use
...
+      measure
+    end
+
+    def self.set_data_criteria_code_list_ids(json, cql_artifacts)
       # Loop over data criteria to search for data criteria that is using a single reference code.
       # Once found set the Data Criteria's 'code_list_id' to our fake oid. Do the same for source data criteria.
       json['data_criteria'].each do |data_criteria_name, data_criteria|
-        # We do not want to replace an existing code_list_id. Skip.
-        unless data_criteria['code_list_id']
+        # We do not want to replace an existing code_list_id. Skip it, unless it is a GUID.
```
Comment (hossenlopp) [inline lib/measures/loading/cql_loader.rb]: This seems like a piece of one time run code. If this updates the GUIDs to be the new hash based IDs... Then it will only be useful until a rebuild_elm is run presumably.

File: lib/measures/loading/cql_loader.rb (hunk 1 above)
Comment (hossenlopp) [reply lib/measures/loading/cql_loader.rb]: Is ALL of the data available for a direct reference code used to create the hash? I don't see anything about code system version here.

File: lib/measures/loading/cql_loader.rb [hunk 3]
```diff
@@ -42,11 +42,28 @@ def self.load_mat_cql_exports(user, zip_file, out_dir, measure_details, vsac_use
...
+        if !data_criteria.key?('code_list_id') || (data_criteria['code_list_id'] && data_criteria['code_list_id'].include?('-'))
```
Comment (hossenlopp) [inline lib/measures/loading/cql_loader.rb]: I think since `drc-` is prepended to the new hash, it would incorrectly identify new hash based IDs as the old GUIDs.

**Review by lukoktonos — APPROVED** [review]
Added tests look good and pass.

Related Bonnie PR working as expected using this branch and was also approved.

**Review by hossenlopp — APPROVED** [review]
